radix-engine/src/transaction/transaction_receipt.rs

```diff
@@ -230,8 +226,16 @@ impl TransactionReceipt {
     }
 
     pub fn output<T: ScryptoDecode>(&self, nth: usize) -> T {
-        scrypto_decode::<T>(&self.expect_commit_success()[nth][..])
-            .expect("Wrong instruction output type!")
+        match &self.expect_commit_success()[nth] {
+            InstructionOutput::Native(native) => {
+                // TODO: Use downcast
+                let value = IndexedScryptoValue::from_typed(&native.as_ref());
+                scrypto_decode::<T>(&value.raw).expect("Wrong instruction output type!")
+            }
+            InstructionOutput::Scrypto(value) => {
+                scrypto_decode::<T>(&value.raw).expect("Wrong instruction output type!")
```

Comment:
maybe it would be useful to print different error message depending on Instruction output kind?
but that's really a minor thing

Reply:
Yup good point